feat(file preview): add bat as plugin

nkxxll opened the pull request:
- add bat as previewer option
- add background color to bat ouput
- add code previewer option to config

references #666

lazysegtree commented on src/internal/model_render.go:
```diff
@@ -701,10 +703,27 @@ func (m *model) filePreviewPanelRender() string {
-			slog.Error("Error render code highlight", "error", err)
-			return box.Render("\n --- " + icon.Error + " Error render code highlight ---")
+		if Config.CodePreviewer == "bat" {
+			batCmd := []string{"bat", itemPath, "-p", "-f"}
```
We are running an external command in the critical path (file_previewing). Please set a timeout so that we dont allow it to get stuck.

Timeout of 0.5second should be fine. 1sec also seems too much for a file preview.

lazysegtree replied:
Please add a comment explaining what are -p and -f options and why are we using them

nkxxll replied:
fair will do 👍

nkxxll replied:
I grepped out the bat preview and left the "if bat" there because taking out the hole thing is kind of ugly because the context has to be passed as arguments and a lot of context is used for the chroma syntax hl

lazysegtree commented on src/internal/model_render.go:
```diff
@@ -701,10 +703,27 @@ func (m *model) filePreviewPanelRender() string {
-		if err != nil {
-			slog.Error("Error render code highlight", "error", err)
-			return box.Render("\n --- " + icon.Error + " Error render code highlight ---")
+		if Config.CodePreviewer == "bat" {
```
Since this is a big chuck of code now, Maybe we should move it out to a separate function. Like getSyntaxHighlightedContent or something.

nkxxll replied:
grepped out the bat part much cleaner now and did the content == "" check before no need to make syntax hl if there is not text to highlight

lazysegtree commented on src/superfile_config/config.toml:
```diff
@@ -47,6 +47,9 @@ transparent_background = false
 # File preview width allow '0' (this mean same as file panel),'x' x must be from 2 to 10 (This means that the width of the file preview will be one xth of the total width.)
 file_preview_width = 0
 #
+# Whether to use the builtin syntax highlighting with chroma or use bat. Values: bat
```
Clarify that leave empty to use chroma and use "bat" to specify bat.

nkxxll replied:
cleared up the comment

lazysegtree commented on src/internal/model_render.go:
```diff
@@ -718,3 +737,11 @@ func (m *model) filePreviewPanelRender() string {
 	return m.commandLine.input.View()
 }
+
+func setBatBackground(input string, background string) string {
```
I will try to go through this part in detail later. Hoping its performant.

lazysegtree commented:
@yorukot Please have a look at this if you have any concerns.

lazysegtree commented:
Also there should be a check that bat exists on the system. It might not, and then maybe we should also print in the file preview that "you have choosen bat, but bat is not there - hence we cant render file preview".

nkxxll commented:
> Also there should be a check that bat exists on the system. It might not, and then maybe we should also print in the file preview that "you have choosen bat, but bat is not there - hence we cant render file preview".

fair point will include that. thanks for the feedback

nkxxll commented:
<img width="1440" alt="image" src="https://github.com/user-attachments/assets/80dedeff-21ae-4210-b5e4-ddd26140ecb8" />
this is how it looks now when you don't have bat installed

lazysegtree commented on src/cmd/main.go:
```diff
@@ -220,6 +222,18 @@ func checkFirstUse() bool {
+		return true
+	}
+	// on ubuntu bat executable is called batcat
+	if _, err := exec.LookPath("bat"); err == nil {
```
Is this a typo ? Was this supposed to be `if _, err := exec.LookPath("batcat"); err == nil {`

nkxxll replied:
Yep, my bad good catch

nkxxll commented:
fixed the typo and squashed the changes also did a little microbenchmark on the background function it seems like the split and join function from the slices lib are pretty optimised. A scanner with custom split is significantly slower (might be because of the bytes conversion overhead). No always creating a new background style gives a little bit more performance because of less allocations but is not a great improvement. If you have any more insights on performance feel free to tell me I am really interested in this side of programming. 😄 

PS: Micro Micro Benchmark results
```
BenchmarkBackground-8                409          29396167 ns/op -> old version slices
BenchmarkBackground2-8                82         147157274 ns/op -> scanner with custom split func
BenchmarkBackground3-8               412          29059363 ns/op -> new version slices without the overhead allocations for the style
```

lazysegtree commented on src/internal/model_render.go:
```diff
@@ -695,26 +699,72 @@ func (m *model) filePreviewPanelRender() string {
+
+	// --plain: use the plain style without line numbers and decorations
+	// --force-colorization: force colorization for non-interactive shell output
+	batCmd := []string{"bat", itemPath, "--plain", "--force-colorization"}
```
@nkxxll As you said ubuntu has "batcat" instead of "bat" . This will not work on ubuntu right ?

nkxxll replied:
That is so true ai hate Ubuntu for that😂 will fix 😉

nkxxll commented:
Fixed the command for ubuntu. I also removed an empty loop this should not do anything the index is not used anywhere the body is empty 🤷‍♂️

lazysegtree commented:
@nkxxll And I forgot one thing. 

Please cut a PR for website update as well to `1.2.1-website-updates` branch.
Add you new config variable in `website/src/content/docs/configure/superfile-config.mdx`

lazysegtree commented:
@nkxxll A strange thing when I dont have bat installed (MacOS), it still loads some file previews without any syntax highlighting. It should not have loaded `flake.lock` preview.


<img width="1140" alt="image" src="https://github.com/user-attachments/assets/c8638b42-6a23-4080-abff-3245cf4577b3" />

<img width="1139" alt="image" src="https://github.com/user-attachments/assets/cdb7327b-ad4e-4b89-997c-f7be180a53f8" />

lazysegtree commented:
@nkxxll Also this doesn't works well in across all themes, for ex : catppuccin-latte

previewer - bat

<img width="1138" alt="image" src="https://github.com/user-attachments/assets/c1aef4f9-0d4b-4ad2-8aaa-1ee110b34b91" />


previewer - ''
<img width="1140" alt="image" src="https://github.com/user-attachments/assets/c44e5873-1925-4ae1-88c2-5a8007c68416" />

lazysegtree commented on src/internal/model_render.go:
```diff
@@ -695,26 +696,68 @@ func (m *model) filePreviewPanelRender() string {
+	ctx, cancel := context.WithTimeout(context.Background(), 500*time.Millisecond)
+	defer cancel()
+
+	cmd := exec.CommandContext(ctx, batCmd[0], append([]string{itemPath}, batCmd[1:]...)...)
```
This needs optimizations. In current code, if the file is of 100K+ lines, we would still only read `previewLines` number of lines . See `readFileContent` function. 

Your current implementation will try to read all 100K lines, and then trim it to 20-30 lines based on the  users terminal size. 

Sorry I am giving many comments one after another, and not all at once. I haven't yet sat down and thoroughly reviewed the changes.

lazysegtree replied:
I thik we should use `--line-range` arg of bat.

nkxxll replied:
yes that is a good idea did not find that option, sorry working on it. No problem, thank you that you keep up with my upsi woopsis 😄. I am happy to help

nkxxll commented:
> @nkxxll A strange thing when I dont have bat installed (MacOS), it still loads some file previews without any syntax highlighting. It should not have loaded `flake.lock` preview.
> 
> <img alt="image" width="1140" src="https://private-user-images.githubusercontent.com/59679977/420645231-c8638b42-6a23-4080-abff-3245cf4577b3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDE1MzkwMDAsIm5iZiI6MTc0MTUzODcwMCwicGF0aCI6Ii81OTY3OTk3Ny80MjA2NDUyMzEtYzg2MzhiNDItNmEyMy00MDgwLWFiZmYtMzI0NWNmNDU3N2IzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzA5VDE2NDUwMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTc0NjY1NDQzMTc0OTQ5MzcyYTg2YzdlNjNhMTE2Y2JjNTMyODhjMWM3MjJiYjQ1ODY3MmQzNjUzYjYyMWI4N2QmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.2Yg_-S-G9SLnGaYsNKfHOd2rgwBPmkn9bnH7GKdcN-g"> <img alt="image" width="1139" src="https://private-user-images.githubusercontent.com/59679977/420645242-cdb7327b-ad4e-4b89-997c-f7be180a53f8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDE1MzkwMDAsIm5iZiI6MTc0MTUzODcwMCwicGF0aCI6Ii81OTY3OTk3Ny80MjA2NDUyNDItY2RiNzMyN2ItYWQ0ZS00Yjg5LTk5N2MtZjdiZTE4MGE1M2Y4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzA5VDE2NDUwMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQyMWYzYzI4OThiY2FlYWI1OTlmOTExMDM5NDliYmQ4M2Q2NmViMmY4Y2E0M2JjODExNDJiYjE5NDg0ODMzZGEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.BGOGD3GaB22FbhEgSEw1Nxa4UPYWoNwSV_7hrS6gwrg">

I let that in because it would also render without highlighting with the builtin preview. So we check whether we know the file type. And if we don't there is no highlighting.

But of course if bat knows the filetype it should render it and if bat is the previewer we should not check if the filetype is known by chroma makes not sense to limit the preview in that way.

I will make it that when we know that the file is a text file we either render with bat or check if we know the filetype in chroma and the render in chroma this makes much more sense...

lazysegtree commented:
> So we check whether we know the file type. And if we don't there is no highlighting.

Ohh. Right. I forgot. This is completely expected. We have a check beforehand and we only highlight syntax of well-known file types.

nkxxll commented:
> @nkxxll Also this doesn't works well in across all themes, for ex : catppuccin-latte
> 
> previewer - bat
> 
> <img alt="image" width="1138" src="https://private-user-images.githubusercontent.com/59679977/420645519-c1aef4f9-0d4b-4ad2-8aaa-1ee110b34b91.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDE1MzkwMDAsIm5iZiI6MTc0MTUzODcwMCwicGF0aCI6Ii81OTY3OTk3Ny80MjA2NDU1MTktYzFhZWY0ZjktMGQ0Yi00YWQyLThhYWEtMWVlMTEwYjM0YjkxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzA5VDE2NDUwMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTJiNTA0ODUyODYxMzRmZDk2NmQwYWQyNzNiY2Q5ZTQyZjVkYjVlNzAzZTVkMDdkOGJhNjEzMzljYzJlZTY3YzYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.4mGGS-pEh4RnF6PtF24se5mm27wKSs3FqpLGprfxWTA">
> previewer - '' <img alt="image" width="1140" src="https://private-user-images.githubusercontent.com/59679977/420645548-c44e5873-1925-4ae1-88c2-5a8007c68416.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDE1MzkwMDAsIm5iZiI6MTc0MTUzODcwMCwicGF0aCI6Ii81OTY3OTk3Ny80MjA2NDU1NDgtYzQ0ZTU4NzMtMTkyNS00YWUxLTg4YzItNWE4MDA3YzY4NDE2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzA5VDE2NDUwMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTFmMDQwNzBmNDA0ZTFhZTc5MzA5ODdhYTg0NTE3OWExODQ0MTg2MTgxN2E5MmE1MzE4NTYwZjU2OTg2ZDBiNWYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.Vyxls7rAcvcqe1IBT3WaRL288Ja3F8nI_1p3z1pci7A">
> 
> bat directly used in terminal
> 
> <img alt="image" width="1005" src="https://private-user-images.githubusercontent.com/59679977/420645843-ee90b642-ca76-4dd7-88c1-0fa2f788a29b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDE1MzkwMDAsIm5iZiI6MTc0MTUzODcwMCwicGF0aCI6Ii81OTY3OTk3Ny80MjA2NDU4NDMtZWU5MGI2NDItY2E3Ni00ZGQ3LTg4YzEtMGZhMmY3ODhhMjliLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzA5VDE2NDUwMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWYxNDUyYzExOGZkZDcyYWEyMzg5OTJjNGJmYjc3ODMyZjk5ZDQwNzBmYTdkY2ZhMzUyYTJiZWRlOWExOTE3M2YmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.t23--4j2FXSmbkwLO0g1avjTynM7VwJQGevCyqi4mOw"> <img alt="image" width="582" src="https://private-user-images.githubusercontent.com/59679977/420645856-1bb9dcba-89f2-4931-9e9b-ae26029259f5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDE1MzkwMDAsIm5iZiI6MTc0MTUzODcwMCwicGF0aCI6Ii81OTY3OTk3Ny80MjA2NDU4NTYtMWJiOWRjYmEtODlmMi00OTMxLTllOWItYWUyNjAyOTI1OWY1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzA5VDE2NDUwMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWU5M2NhY2E1ZDU4YjMyYzQwYmFmOTkzZDU0NmNhM2UzZDMxYWI4MTU1MDQzOWViMTk3ZWY0OTE4NmVhMTY3MmUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.IZ5P_TVUD7wBi0zlJwQT9A6mathWC8XenNtWFByU0hk">

I think i know that is the reason for that you would have to configure bat to use the dark or light variant of the theme depending on your superfile background color. the first image uses the dark variant probably but should use the light variant. I might have to toggle 

> @nkxxll Also this doesn't works well in across all themes, for ex : catppuccin-latte
> 
> previewer - bat
> 
> <img alt="image" width="1138" src="https://private-user-images.githubusercontent.com/59679977/420645519-c1aef4f9-0d4b-4ad2-8aaa-1ee110b34b91.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDE1MzkwMDAsIm5iZiI6MTc0MTUzODcwMCwicGF0aCI6Ii81OTY3OTk3Ny80MjA2NDU1MTktYzFhZWY0ZjktMGQ0Yi00YWQyLThhYWEtMWVlMTEwYjM0YjkxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzA5VDE2NDUwMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTJiNTA0ODUyODYxMzRmZDk2NmQwYWQyNzNiY2Q5ZTQyZjVkYjVlNzAzZTVkMDdkOGJhNjEzMzljYzJlZTY3YzYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.4mGGS-pEh4RnF6PtF24se5mm27wKSs3FqpLGprfxWTA">
> previewer - '' <img alt="image" width="1140" src="https://private-user-images.githubusercontent.com/59679977/420645548-c44e5873-1925-4ae1-88c2-5a8007c68416.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDE1MzkwMDAsIm5iZiI6MTc0MTUzODcwMCwicGF0aCI6Ii81OTY3OTk3Ny80MjA2NDU1NDgtYzQ0ZTU4NzMtMTkyNS00YWUxLTg4YzItNWE4MDA3YzY4NDE2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzA5VDE2NDUwMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTFmMDQwNzBmNDA0ZTFhZTc5MzA5ODdhYTg0NTE3OWExODQ0MTg2MTgxN2E5MmE1MzE4NTYwZjU2OTg2ZDBiNWYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.Vyxls7rAcvcqe1IBT3WaRL288Ja3F8nI_1p3z1pci7A">
> 
> bat directly used in terminal
> 
> <img alt="image" width="1005" src="https://private-user-images.githubusercontent.com/59679977/420645843-ee90b642-ca76-4dd7-88c1-0fa2f788a29b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDE1MzkwMDAsIm5iZiI6MTc0MTUzODcwMCwicGF0aCI6Ii81OTY3OTk3Ny80MjA2NDU4NDMtZWU5MGI2NDItY2E3Ni00ZGQ3LTg4YzEtMGZhMmY3ODhhMjliLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzA5VDE2NDUwMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWYxNDUyYzExOGZkZDcyYWEyMzg5OTJjNGJmYjc3ODMyZjk5ZDQwNzBmYTdkY2ZhMzUyYTJiZWRlOWExOTE3M2YmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.t23--4j2FXSmbkwLO0g1avjTynM7VwJQGevCyqi4mOw"> <img alt="image" width="582" src="https://private-user-images.githubusercontent.com/59679977/420645856-1bb9dcba-89f2-4931-9e9b-ae26029259f5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDE1MzkwMDAsIm5iZiI6MTc0MTUzODcwMCwicGF0aCI6Ii81OTY3OTk3Ny80MjA2NDU4NTYtMWJiOWRjYmEtODlmMi00OTMxLTllOWItYWUyNjAyOTI1OWY1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzA5VDE2NDUwMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWU5M2NhY2E1ZDU4YjMyYzQwYmFmOTkzZDU0NmNhM2UzZDMxYWI4MTU1MDQzOWViMTk3ZWY0OTE4NmVhMTY3MmUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.IZ5P_TVUD7wBi0zlJwQT9A6mathWC8XenNtWFByU0hk">

I might be wrong but it does not look like bat is using catppuccin-latte theme in spf how do you have bat configured? If it uses ansi colors and your terminal background is dark starting spf bat could auto detects a theme based on the background color which might lead to the dark color scheme in the light spf theme. There is a point that with the wrong bat config the preview could not work properly.
But at the moment I don't have an idea of how to handle that open for ideas though.

lazysegtree commented:
@nkxxll My setup
- MacOS 14 iterm2 with dark mode
- spf with catapaccin theme

<img width="1145" alt="image" src="https://github.com/user-attachments/assets/1ecf2f94-a2bd-4e5a-9b2e-8878799484f6" />

<img width="1147" alt="image" src="https://github.com/user-attachments/assets/13ac5d24-c95b-48cb-a23f-ecbb5354d359" />

nkxxll commented:
> @nkxxll My setup
> 
> * MacOS 14 iterm2 with dark mode
> * spf with catapaccin theme
> 
> <img alt="image" width="1145" src="https://private-user-images.githubusercontent.com/59679977/420942227-1ecf2f94-a2bd-4e5a-9b2e-8878799484f6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDE2Mzc0NTcsIm5iZiI6MTc0MTYzNzE1NywicGF0aCI6Ii81OTY3OTk3Ny80MjA5NDIyMjctMWVjZjJmOTQtYTJiZC00ZTVhLTliMmUtODg3ODc5OTQ4NGY2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzEwVDIwMDU1N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTNjMzQ1ZDAzODJiZTg1M2FmM2Y1MTU0YzAwMmQ5OWJiMWQ3ZDkyZmIwMmU2NjlhNzNlOTk4MDAyZWI1YmUxZWMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.MIse7xibkelH31qqtiWiq-oUqe9_23Ink78pRtHLGms"> <img alt="image" width="1147" src="https://private-user-images.githubusercontent.com/59679977/420942443-13ac5d24-c95b-48cb-a23f-ecbb5354d359.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDE2Mzc0NTcsIm5iZiI6MTc0MTYzNzE1NywicGF0aCI6Ii81OTY3OTk3Ny80MjA5NDI0NDMtMTNhYzVkMjQtYzk1Yi00OGNiLWEyM2YtZWNiYjUzNTRkMzU5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzEwVDIwMDU1N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWJhZGQ0OWJhMDJlM2MwOWQxZmQ1ZmEzNjVlODAzOThjYjZhOWE3ZWI2Nzk0NDU2N2ExZWZlNWQxYTEzNjFlOTAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.7zR_mxczmIlcOUQqkwwbxsXguHTH125ggnUeV64-mGw">
> Issue is that ansi chroma highlighter adjusts the colors based on my spf theme, which is probably correct way to do it. while bat highlighter does not, and it prioritised the terminal colors

Yeah this is probably more a feature than a bug, by default bat chooses a theme based on the background color. If you want to have another theme you have to configure bat that way. Only thing that would fix that is that we give the option to pass flags to bat for preview you could use the `--theme` option for that. Instead of previewer we would also have bat cmd or previewer command. When we choose the theme corresponding to the spf theme we in a way destroy the flexibility we wanted to give the user with bat syntax highlighting right? Whats your opinion on that?

lazysegtree commented:
"When we choose the theme corresponding to the spf theme we in a way destroy the flexibility we wanted to give the user with bat syntax highlighting right?"
Yes makes sense.

lazysegtree commented:
@nkxxll For PR to website, I meant this . I have created it. You dont need to do it now.

https://github.com/yorukot/superfile/pull/701

lazysegtree commented on website/src/content/docs/configure/superfile-config.mdx:
```diff
@@ -159,6 +159,11 @@ This setting is an integer.
 `X` must be from 2 to 10.
 :::
 
+- ###### code_previewer
```
Website changes cannot be merged to master before the new release is published. 
Created a separate PR for this
https://github.com/yorukot/superfile/pull/701

nkxxll replied:
Ohh makes sense, thank you!

lazysegtree commented on src/superfile_config/config.toml:
```diff
@@ -37,6 +37,9 @@ case_sensitive_sort = false
 # ================   Style =================
+#
+# Whether to use the builtin syntax highlighting with chroma or use bat. Values: "" for builtin chroma, "bat" for bat
+code_previewer = ''
```
@nkxxll Used single quotes for consistency, and moved it to correct order.

Earlier it was between `file_preview_width` and `sidebar_width`. Out of place.

nkxxll replied:
fair 😉 thanks

nkxxll commented:
Squashed the changes do I have to anything else? @lazysegtree
